Repo scaffold

mikekeke opened the pull request:
Repo scaffold:
- added: flake.nix
- added: basic repo structure and cabal files
- added: `plutus-tx` dependency

mikekeke committed:
added: nix flake and scaffold haskell project

mikekeke committed:
basic check that stuff builds and runs with plutus-tx

mikekeke commented on milestone34.cabal:
```diff
@@ -0,0 +1,45 @@
+cabal-version:   3.0
+name:            milestone34
```
I've tried to name it `milestone-3-4` but got cabal parsing error.

kozross replied:
The exact name doesn't matter very much - this will work OK.

kozross commented on milestone34.cabal:
```diff
@@ -0,0 +1,45 @@
+common warnings
+  ghc-options: -Wall
+
+library
+  import:           warnings
+  exposed-modules:  MyLib
+
+  -- other-modules:
+  -- other-extensions:
+  build-depends:
+    , base         ^>=4.18.2.0
+    , plutus-tx
+
+  hs-source-dirs:   src
+  default-language: Haskell2010
```
PlutusTx requires a bunch of GHC flags and extensions to properly work, and it'd be good if we could also add these.

mikekeke replied:
Added according to [this doc](https://plutus.readthedocs.io/en/latest/extensions-flags-pragmas.html#), plus `-fplugin-opt PlutusTx.Plugin:defer-errors`

chfanghr committed:
setup hercules ci

chfanghr commented on flake.nix:
```diff
@@ -0,0 +1,62 @@
+{
+  description = "milestone34";
+  nixConfig = {
```
I think it's a good idea to remove this `nixConfig` block altogether and put some instructions in the README instead. 

Quote from @brainrake:

> Don't do this, it's insecure.
> The only way to use trusted-public-keys from nix flake is if you put your user in trusted-users, which gives them > passwordless root.

kozross replied:
Yeah, exactly this. I got told the same thing yesterday.

mikekeke replied:
Got it. I thought just not having `trusted-public-keys` there is enough.

Does anybody know how critical is `allow-import-from-derivation = "true";` here? (since this option was never enabled  in my `nix.conf` and I was able to run `nix develop` w/o it)

mikekeke replied:
Seems like `allow-import-from-derivation = "true";` is `true` by default: https://nixos.org/manual/nix/stable/command-ref/conf-file#conf-allow-import-from-derivation. So I removed whole `nixConfig` section.

chfanghr committed:
copy pre-commit setup from plutus-conformance

chfanghr committed:
make test pass

chfanghr committed:
run cabal-fmt

mikekeke committed:
added: PlutusTx-specific GHC  flags and options

mikekeke committed:
dev environment adjustments
- removed: `nixConfig`  attribute set from `flake.nix`
- added: instructions on adding plutonomicon binary cache to readme

chfanghr committed:
Merge pull request #5 from mlabs-haskell/connor/ci

CI Setup

chfanghr commented on flake.nix:
```diff
@@ -0,0 +1,85 @@
+                haskell-language-server = { };
+                hlint = { };
+                cabal-fmt = { };
+                fourmolu = { };
```
We're using `ormolu` now.

mikekeke replied:
Changed

chfanghr commented on flake.nix:
```diff
@@ -0,0 +1,85 @@
+            deadnix.enable = true;
+            statix.enable = true;
+            cabal-fmt.enable = true;
+            # TODO(chfanghr): Configuration for ormolu
```
Please remove this TODO, no configuration is required.

mikekeke committed:
fixing pre-commit hook errors, `flake.nix` comments cleanup

mikekeke committed:
dev shell `fourmolu` -> `ormolu`
